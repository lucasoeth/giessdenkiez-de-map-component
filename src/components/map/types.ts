export interface Generic {
  [key: string]: any;
}

export interface ViewState {
  bearing: number;
  latitude: number;
  longitude: number;
  maxZoom: number;
  minZoom: number;
  pitch: number;
  zoom: number;
}

export interface Feature {
  type: string;
  geometry: {
    type: string;
    coordinates: number[] | number[][];
  };
  properties: {
    id: string | number;
  };
}

export interface TreeFeature extends Feature {
  properties: {
    id: string;
    radolan_sum?: number;
    age: number;
  };
}

export interface PumpFeature extends Feature {
  properties: {
    id: number;
    'addr:full': string;
    check_date: string;
    image: string;
    'pump:status': string;
    'pump:style': string;
  };
}

export interface RainFeature extends Feature {
  properties: {
    id: number;
    data: number[];
  };
}

export interface FeatureCollection {
  type: string;
  features: TreeFeature[] | PumpFeature[] | RainFeature[];
}

export type OnViewStateChange = (e: unknown) => void;
export type SetSelectedTree = (id: string) => void;
export type SetLayer = (layer: string) => void;

export interface HoveredPump {
  message: string;
  pointer: number[];
}

export type SetHoveredPump = (info: HoveredPump) => void;

//Color.ts
export type RGBAColor = [number, number, number, number];

export interface Color extends Generic {
  name: string;
  rgba: RGBAColor;
  hex: string;
}
