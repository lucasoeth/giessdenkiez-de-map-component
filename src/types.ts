import { GeoJsonLayer, GeoJsonLayerProps } from '@deck.gl/layers';
//ViewState
export interface ViewState {
  bearing: number;
  latitude: number;
  longitude: number;
  maxZoom: number;
  minZoom: number;
  pitch: number;
  zoom: number;
}
export type OnViewStateChange = (e: unknown) => void;
//GeoJson
export interface FeatureCollection {
  type: string;
  features: TreeFeature[] | PumpFeature[] | RainFeature[];
}
export interface Feature {
  type: string;
  geometry: {
    type: string;
    coordinates: number[] | [number, number][][];
  };
  properties: {
    id: string | number;
  };
}
//Tree
export interface TreeFeature extends Feature {
  properties: {
    id: string;
    radolan_sum?: number;
    age: number;
  };
}
export type SetSelectedTree = (id: string) => void;
//Pump
export interface PumpFeature extends Feature {
  properties: {
    id: number;
    'addr:full': string;
    check_date: string;
    image: string;
    'pump:status': string;
    'pump:style': string;
  };
}
export interface HoveredPump {
  message: string;
  pointer: number[];
}
export type SetHoveredPump = (info: HoveredPump) => void;
//Rain
export interface RainFeature extends Feature {
  properties: {
    id: number;
    data: number[];
  };
}
//Watered
export interface CommunityData {
  [key: string]: {
    watered: boolean;
    adopted: boolean;
  };
}
//Layer
export type SetLayer = (layer: string) => void;

export type Layer<FeatureType> = GeoJsonLayer<
  FeatureType,
  GeoJsonLayerProps<FeatureType>
>;

//App
export interface WhichLayer {
  showTrees: boolean;
  showWatered: boolean;
  showAdopted: boolean;
  showPumps: boolean;
  showRain: boolean;
}

export type WhichLayerReducer = (
  previousWhichLayer: WhichLayer,
  layer: string
) => WhichLayer;

export type AgeRange = [number, number];
