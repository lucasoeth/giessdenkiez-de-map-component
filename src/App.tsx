import React from 'react';
// import logo from './logo.svg';
import './App.css';
import { ThemeProvider } from 'styled-components';
import { theme } from './assets/theme';

import Map from './components/map';

function App() {
  return (
    <div className='App'>
      <ThemeProvider theme={theme}>
        <Map />
      </ThemeProvider>
    </div>
  );
}

export default App;
