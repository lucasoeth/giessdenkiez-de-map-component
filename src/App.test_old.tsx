import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  // render((<App />) as JSX.Element);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
