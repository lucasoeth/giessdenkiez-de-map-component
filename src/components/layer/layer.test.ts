import { getLayers, getPumpLayer, getRainLayer } from '.';

import { FeatureCollection } from 'src/types';

const test_trees: FeatureCollection = {
  type: 'FeatureCollection',
  features: [
    {
      type: 'Feature',
      geometry: { type: 'Point', coordinates: [13.50213, 52.645680000000006] },
      properties: { id: 'nt240620302094', radolan_sum: 43.9, age: 2 },
    },
  ],
};

const test_pumps: FeatureCollection = {
  type: 'FeatureCollection',
  features: [
    {
      id: '0',
      type: 'Feature',
      properties: {
        'addr:full': 'Schragenfeldstraße 25',
        check_date: 'unbekannt',
        id: 352734260,
        image: 'File:Plumpe 11 Marzahn Schragenfeldstraße-Bäckerpfuhl (8).jpg',
        'pump:status': 'unbekannt',
        'pump:style': 'Borsig',
      },
      geometry: {
        type: 'Point',
        coordinates: [13.564219, 52.5398067],
      },
    },
  ],
};

const test_rain: FeatureCollection = {
  type: 'FeatureCollection',
  features: [
    {
      type: 'Feature',
      geometry: {
        type: 'Polygon',
        coordinates: [
          [
            [13.4971096103036, 52.6543762343589],
            [13.5042366641048, 52.6541120206218],
            [13.5033652084477, 52.64547732806],
            [13.4891325166817, 52.6460049371748],
            [13.4900002353473, 52.654638719058],
            [13.4971096103036, 52.6543762343589],
          ],
        ],
      },
      properties: {
        id: 1,
        data: [571],
      },
    },
  ],
};

const test_communityData = {
  _00l734zv4: {
    watered: true,
    adopted: false,
  },
};

const emptyFunction = function () {
  return 0;
};

describe('DeckGL Layer Generation Functions', () => {
  // test('getTreeLayer(showPumps=true, pumps=test_pumps, setHoveredPump=emptyFunction)', () => {
  //   expect(getTreeLayer(true, test_pumps, emptyFunction)).toMatchSnapshot();
  // });
  test('getPumpLayer(showPumps=true, pumps=test_pumps, setHoveredPump=emptyFunction)', () => {
    expect(getPumpLayer(true, test_pumps, emptyFunction)).toMatchSnapshot();
  });
  test('getPumpLayer(showPumps=false, pumps=test_pumps, setHoveredPump=emptyFunction)', () => {
    expect(getPumpLayer(false, test_pumps, emptyFunction)).toMatchSnapshot();
  });
  test('getRainLayer(showRain=true, rain=test_rain)', () => {
    expect(getRainLayer(true, test_rain)).toMatchSnapshot();
  });
  test('getRainLayer(showRain=false, rain=test_rain)', () => {
    expect(getRainLayer(false, test_rain)).toMatchSnapshot();
  });
});
