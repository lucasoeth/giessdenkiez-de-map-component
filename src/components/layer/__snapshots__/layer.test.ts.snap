// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckGL Layer Generation Functions getPumpLayer(showPumps=false, pumps=test_pumps, setHoveredPump=emptyFunction) 1`] = `
GeoJsonLayer {
  "context": null,
  "count": 1,
  "id": "pumps",
  "internalState": null,
  "lifecycle": "Awaiting state",
  "parent": null,
  "props": Object {
    "extruded": true,
    "filled": true,
    "getFillColor": [Function],
    "getLineColor": Array [
      0,
      0,
      0,
      200,
    ],
    "getLineWidth": 3,
    "getRadius": 9,
    "id": "pumps",
    "lineWidthMinPixels": 1.5,
    "onHover": [Function],
    "opacity": 1,
    "pickable": true,
    "pointRadiusMinPixels": 4,
    "stroked": true,
    "visible": false,
    Symbol(component): [Circular],
    Symbol(asyncPropOriginal): Object {},
    Symbol(asyncPropResolved): Object {
      "data": Array [
        Object {
          "geometry": Object {
            "coordinates": Array [
              13.564219,
              52.5398067,
            ],
            "type": "Point",
          },
          "id": "0",
          "properties": Object {
            "addr:full": "Schragenfeldstraße 25",
            "check_date": "unbekannt",
            "id": 352734260,
            "image": "File:Plumpe 11 Marzahn Schragenfeldstraße-Bäckerpfuhl (8).jpg",
            "pump:status": "unbekannt",
            "pump:style": "Borsig",
          },
          "type": "Feature",
        },
      ],
    },
  },
  "state": null,
}
`;

exports[`DeckGL Layer Generation Functions getPumpLayer(showPumps=true, pumps=test_pumps, setHoveredPump=emptyFunction) 1`] = `
GeoJsonLayer {
  "context": null,
  "count": 0,
  "id": "pumps",
  "internalState": null,
  "lifecycle": "Awaiting state",
  "parent": null,
  "props": Object {
    "extruded": true,
    "filled": true,
    "getFillColor": [Function],
    "getLineColor": Array [
      0,
      0,
      0,
      200,
    ],
    "getLineWidth": 3,
    "getRadius": 9,
    "id": "pumps",
    "lineWidthMinPixels": 1.5,
    "onHover": [Function],
    "opacity": 1,
    "pickable": true,
    "pointRadiusMinPixels": 4,
    "stroked": true,
    "visible": true,
    Symbol(component): [Circular],
    Symbol(asyncPropOriginal): Object {},
    Symbol(asyncPropResolved): Object {
      "data": Array [
        Object {
          "geometry": Object {
            "coordinates": Array [
              13.564219,
              52.5398067,
            ],
            "type": "Point",
          },
          "id": "0",
          "properties": Object {
            "addr:full": "Schragenfeldstraße 25",
            "check_date": "unbekannt",
            "id": 352734260,
            "image": "File:Plumpe 11 Marzahn Schragenfeldstraße-Bäckerpfuhl (8).jpg",
            "pump:status": "unbekannt",
            "pump:style": "Borsig",
          },
          "type": "Feature",
        },
      ],
    },
  },
  "state": null,
}
`;

exports[`DeckGL Layer Generation Functions getRainLayer(showRain=false, rain=test_rain) 1`] = `
GeoJsonLayer {
  "context": null,
  "count": 3,
  "id": "rain",
  "internalState": null,
  "lifecycle": "Awaiting state",
  "parent": null,
  "props": Object {
    "extruded": true,
    "filled": true,
    "getElevation": 1,
    "getFillColor": [Function],
    "getLineColor": Array [
      0,
      0,
      0,
      200,
    ],
    "getLineWidth": 3,
    "getRadius": 9,
    "id": "rain",
    "lineWidthMinPixels": 1.5,
    "opacity": 0.95,
    "pickable": true,
    "pointRadiusMinPixels": 4,
    "stroked": false,
    "visible": false,
    "wireframe": true,
    Symbol(component): [Circular],
    Symbol(asyncPropOriginal): Object {},
    Symbol(asyncPropResolved): Object {
      "data": Array [
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  13.4971096103036,
                  52.6543762343589,
                ],
                Array [
                  13.5042366641048,
                  52.6541120206218,
                ],
                Array [
                  13.5033652084477,
                  52.64547732806,
                ],
                Array [
                  13.4891325166817,
                  52.6460049371748,
                ],
                Array [
                  13.4900002353473,
                  52.654638719058,
                ],
                Array [
                  13.4971096103036,
                  52.6543762343589,
                ],
              ],
            ],
            "type": "Polygon",
          },
          "properties": Object {
            "data": Array [
              571,
            ],
            "id": 1,
          },
          "type": "Feature",
        },
      ],
    },
  },
  "state": null,
}
`;

exports[`DeckGL Layer Generation Functions getRainLayer(showRain=true, rain=test_rain) 1`] = `
GeoJsonLayer {
  "context": null,
  "count": 2,
  "id": "rain",
  "internalState": null,
  "lifecycle": "Awaiting state",
  "parent": null,
  "props": Object {
    "extruded": true,
    "filled": true,
    "getElevation": 1,
    "getFillColor": [Function],
    "getLineColor": Array [
      0,
      0,
      0,
      200,
    ],
    "getLineWidth": 3,
    "getRadius": 9,
    "id": "rain",
    "lineWidthMinPixels": 1.5,
    "opacity": 0.95,
    "pickable": true,
    "pointRadiusMinPixels": 4,
    "stroked": false,
    "visible": true,
    "wireframe": true,
    Symbol(component): [Circular],
    Symbol(asyncPropOriginal): Object {},
    Symbol(asyncPropResolved): Object {
      "data": Array [
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  13.4971096103036,
                  52.6543762343589,
                ],
                Array [
                  13.5042366641048,
                  52.6541120206218,
                ],
                Array [
                  13.5033652084477,
                  52.64547732806,
                ],
                Array [
                  13.4891325166817,
                  52.6460049371748,
                ],
                Array [
                  13.4900002353473,
                  52.654638719058,
                ],
                Array [
                  13.4971096103036,
                  52.6543762343589,
                ],
              ],
            ],
            "type": "Polygon",
          },
          "properties": Object {
            "data": Array [
              571,
            ],
            "id": 1,
          },
          "type": "Feature",
        },
      ],
    },
  },
  "state": null,
}
`;
